lib/suse/connect/zypper.rb

```diff
@@ -47,7 +47,8 @@ def refresh
 
         # Returns an array of hashes of all available repositories
         def repositories
-          zypper_out = call('--xmlout --non-interactive repos -d', false)
+          # Don't fail when zypper exits with 6 (no repositories)
+          zypper_out = call('--xmlout --non-interactive repos -d', false, [0, 6])
```

Comment:
Can I suggest having constants for all the Zypper exit codes? It can be a separate module or we can add them to the `SUSE::Connect::ZypperError` class.

Places like this will become a little bit more verbose, but much more readable:
```ruby
zypper_out = call('--xmlout --non-interactive repos -d', false, [Zypper::ExitCode::OK, Zypper::ExitCode::NO_REPOS])
```
and also allow us to implement more human-readable zypper errors reporting at some point in the future.